From dc8aa78c6aa7340b578f8f75274fc7fc8f80d3d7 Mon Sep 17 00:00:00 2001
From: Victor Yudin <victor@yudin.org>
Date: Wed, 7 Feb 2018 20:09:51 -0500
Subject: [PATCH 1/2] Weak function _ReadPlugInfoObject

It will allow us to use internal plugInfo.json and get rid of garbage on
hard drive.
---
 pxr/base/lib/plug/info.cpp | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pxr/base/lib/plug/info.cpp b/pxr/base/lib/plug/info.cpp
index 53b0ff6a..344bcb19 100644
--- a/pxr/base/lib/plug/info.cpp
+++ b/pxr/base/lib/plug/info.cpp
@@ -144,8 +144,8 @@ _AddPlugin(
 
 // Return the plug info in pathname into result.  Returns true if the
 // file could be opened.
-bool
-_ReadPlugInfoObject(const std::string& pathname, JsObject* result)
+extern "C" bool
+_DefaultReadPlugInfoObject(const std::string& pathname, JsObject* result)
 {
     result->clear();
 
@@ -193,6 +193,9 @@ _ReadPlugInfoObject(const std::string& pathname, JsObject* result)
     return true;
 }
 
+extern "C" bool (*_ReadPlugInfoObject)(const std::string& pathname, JsObject* result);
+#pragma comment(linker, "/alternatename:_ReadPlugInfoObject=_DefaultReadPlugInfoObject")
+
 void
 _ReadPlugInfoWithWildcards(_ReadContext* context, const std::string& pathname);
 
-- 
2.13.0.windows.1

